### PR TITLE
test: improve test coverage

### PR DESCRIPTION
**Pull request by TJM**
Add a few error state tests to improve test coverage on config/rotate

* `path_config_rotate.go`
  * master: 77.1% (overall 62.4%)
  * this branch: 88.6% (overall 65%)

### CODE REVIEW

**Comment by TJM** [comment]
We have all the "positive" test cases covered, but I am not sure how to create some of the errors. If you have ideas, let me know. I am going to tackle /config/admin next.

**Comment by TJM** [comment]
* `path_config.go`
  * master: 68.1% (62.4% overall)
  * this branch: 91.2% (68.2% overall)

File: path_config.go [hunk 1]
```diff
@@ -104,7 +104,6 @@ func (b *backend) pathConfigUpdate(ctx context.Context, req *logical.Request, da
 	}
 
 	if val, ok := data.GetOk("use_expiring_tokens"); ok {
-		b.Logger().Warn("config update use_expiring_tokens", "use_expiring_tokens", val)
```
Comment (TJM) [inline path_config.go:107]: I noticed this while testing the expiring_tokens, this was in here for debug, removed.

**Comment by alexhung** [comment]
> Learned something... Screenshot 2023-04-26 at 8 46 13 AM
> 
> We don't need this case statement to check for `bool`, because no matter what we submit, it comes out as a string? although this feels like a bug to me :)

Most likely because the attribute is defined as `TypeString` (https://github.com/jfrog/artifactory-secrets-plugin/blob/master/path_config.go#L33) 😄 

Instead we should use `TypeBool`.

File: path_config_rotate_test.go [hunk 2]
```diff
@@ -37,3 +66,31 @@ func (e *accTestEnv) PathConfigRotateWithDetails(t *testing.T) {
...
+	// Forcibly set a bad token
+	entry, err := logical.StorageEntryJSON("config/admin", adminConfiguration{
+		AccessToken:    "bogus.token",
+		ArtifactoryURL: e.URL,
+	})
+	assert.NoError(t, err)
+	err = e.Storage.Put(e.Context, entry)
+	assert.NoError(t, err)
+	resp, err := e.update("config/rotate", testData{})
+	// assert.Contains(t, resp.Data["error"], "error parsing existing AccessToken")
```
Comment (alexhung) [inline path_config_rotate_test.go:93]: Is this still needed?

**Review by alexhung — CHANGES_REQUESTED** [review]
LGTM. Just the commented out line of code.

File: path_config_rotate_test.go (hunk 2 above)
Comment (TJM) [reply path_config_rotate_test.go:93]: This is a matter of opinion, I think that it should contain the error, but it does not. I want to adjust this in a future MR.. trying to not change a bunch of code while writing the tests ;)

**Comment by TJM** [comment]
> > Learned something... Screenshot 2023-04-26 at 8 46 13 AM
> > We don't need this case statement to check for `bool`, because no matter what we submit, it comes out as a string? although this feels like a bug to me :)
> 
> Most likely because the attribute is defined as `TypeString` (https://github.com/jfrog/artifactory-secrets-plugin/blob/master/path_config.go#L33) 😄
> 
> Instead we should use `TypeBool`.

Created a new issue for this.

**Review by alexhung — APPROVED** [review]

File: path_config_rotate_test.go (hunk 2 above)
Comment (TJM) [reply path_config_rotate_test.go:93]: See #92

**Pull request merged**